hummingbot/connector/derivative/binance_perpetual/binance_perpetual_api_order_book_data_source.py

```diff
@@ -59,59 +87,109 @@ async def get_last_traded_prices(cls, trading_pairs: List[str], domain: str = CO
+                url=url,
+                params=params,
+            )
+            resp = await rest_assistant.call(request=request)
+            resp_json = await resp.json()
+            return float(resp_json["lastPrice"])
 
     @classmethod
-    def _get_throttler_instance(cls) -> AsyncThrottler:
-        return AsyncThrottler(CONSTANTS.RATE_LIMITS)
+    async def trading_pair_symbol_map(
+        cls, domain: Optional[str] = CONSTANTS.DOMAIN, throttler: Optional[AsyncThrottler] = None
+    ) -> Dict[str, str]:
+        if not cls._trading_pair_symbol_map:
+            await cls.init_trading_pair_symbols(domain, throttler)
+        return cls._trading_pair_symbol_map
+
+    @classmethod
+    async def init_trading_pair_symbols(
+        cls, domain: str = CONSTANTS.DOMAIN, throttler: Optional[AsyncThrottler] = None
+    ):
+        """Initialize _trading_pair_symbol_map class variable"""
+        api_factory = utils.build_api_factory()
+        rest_assistant = await api_factory.get_rest_assistant()
 
-    @staticmethod
-    async def fetch_trading_pairs(domain: str = CONSTANTS.DOMAIN, throttler: Optional[AsyncThrottler] = None) -> List[str]:
         url = utils.rest_url(path_url=CONSTANTS.EXCHANGE_INFO_URL, domain=domain)
-        throttler = throttler or BinancePerpetualAPIOrderBookDataSource._get_throttler_instance()
+        throttler = throttler or cls._get_throttler_instance()
         async with throttler.execute_task(limit_id=CONSTANTS.EXCHANGE_INFO_URL):
-            async with aiohttp.ClientSession() as client:
-                async with client.get(url=url, timeout=10) as response:
-                    if response.status == 200:
-                        data = await response.json()
-                        # fetch d["pair"] for binance perpetual
-                        raw_trading_pairs = [d["pair"] for d in data["symbols"] if d["status"] == "TRADING"]
-                        trading_pair_targets = [
-                            f"{d['baseAsset']}-{d['quoteAsset']}" for d in data["symbols"] if d["status"] == "TRADING"
-                        ]
-                        trading_pair_list: List[str] = []
-                        for raw_trading_pair, pair_target in zip(raw_trading_pairs, trading_pair_targets):
-                            trading_pair = utils.convert_from_exchange_trading_pair(raw_trading_pair)
-                            if trading_pair is not None and trading_pair == pair_target:
-                                trading_pair_list.append(trading_pair)
-                        return trading_pair_list
-        return []
+            request = RESTRequest(
+                method=RESTMethod.GET,
+                url=url,
+            )
+            response = await rest_assistant.call(request=request, timeout=10)
+
+            if response.status == 200:
+                data = await response.json()
+                # fetch d["pair"] for binance perpetual
+                cls._trading_pair_symbol_map = {
```

Comment:
I talked about it with Abel and we agreed to keep it the old way, because it would require the bidict library and it's not a scope of the refactoring, the description of the task refers to Ndax and Beaxy.